backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/optimizations/Devirtualization.kt

```diff
@@ -994,20 +993,32 @@ internal object Devirtualization {
                         .asSequence()
                         .filter { it.key.irCallSite != null }
                         .associate { it.key.irCallSite!! to it.value }
-        Devirtualization.devirtualize(irModule, context, moduleDFG, externalModulesDFG, devirtualizedCallSites)
+        devirtualize(irModule, context, moduleDFG, externalModulesDFG, devirtualizedCallSites)
+        removeRedundantCoercions(irModule, context, moduleDFG, externalModulesDFG)
         return AnalysisResult(devirtualizationAnalysisResult)
     }
 
+    /**
+     * TODO: JVM inliner crashed on attempt inline this function from transform.kt with:
+     *  j.l.IllegalStateException: Couldn't obtain compiled function body for
+     *  public inline fun <reified T : org.jetbrains.kotlin.ir.IrElement> kotlin.collections.MutableList<T>.transform...
+     */
+    private inline fun <reified T : IrElement> MutableList<T>.transform(transformation: (T) -> IrElement) {
+        forEachIndexed { i, item ->
+            set(i, transformation(item) as T)
+        }
+    }
+
+    // TODO: do it more reliably.
+    private fun IrExpression.isBoxOrUnboxCall() =
+            (this is IrCall && symbol.owner.name.asString().let { it == "<box>" || it == "<unbox>" })
+                    || (this is IrPrivateFunctionCall && dfgSymbol.name?.let { it.contains("<box>") || it.contains("<unbox>") } == true)
```

Comment:
Somewhat ugly logic. Can we list all box/unbox function is set and check with `contains`?

Reply:
I left "TODO" exactly because of this.